### PR TITLE
Fix MissingFieldException in offline mode

### PR DESCRIPTION
**Pull request by Blaizzy**
## What does this PR do?

It fixes MissingFieldException in offline mode reported here: https://github.com/neptune-ai/neptune-client/issues/893

### Does your PR introduce any breaking changes? If yes, please list them.

No.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Yes, I did 😃 !

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

File: src/pytorch_lightning/CHANGELOG.md [hunk 1]
```diff
@@ -267,6 +267,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Fixed an attribute error when running the tuner together with the `StochasticWeightAveraging` callback ([#14836](https://github.com/Lightning-AI/lightning/pull/14836))
 
+- Fixed MissingFieldException in offline mode for the `NeptuneLogger()`([#14919](https://github.com/Lightning-AI/lightning/pull/14919))
```
Comment (Raalsky) [inline src/pytorch_lightning/CHANGELOG.md:270]: That's a little one but there should be a space before URL 🧐

File: src/pytorch_lightning/loggers/neptune.py [hunk 2]
```diff
@@ -250,12 +249,12 @@ def __init__(
             self._run_instance[_INTEGRATION_VERSION_KEY] = pl.__version__
 
     def _retrieve_run_data(self) -> None:
-        try:
-            assert self._run_instance is not None
-            self._run_instance.wait()
+        assert self._run_instance is not None
+        self._run_instance.wait()
+        if self._run_instance.exists("sys/id"):
```
Comment (Raalsky) [inline src/pytorch_lightning/loggers/neptune.py:254]: Can we place a newline before if statement?

File: tests/tests_pytorch/loggers/test_neptune.py [hunk 3]
```diff
@@ -40,7 +40,23 @@ def create_neptune_mock():
     Mostly due to fact, that windows tests were failing with MagicMock based strings, which were used to create local
     directories in FS.
     """
-    return MagicMock(init=MagicMock(return_value=MagicMock(__getitem__=MagicMock(side_effect=fetchable_paths))))
+    return MagicMock(
```
Comment (Raalsky) [inline tests/tests_pytorch/loggers/test_neptune.py:43]: We can merge both functions (`create_neptune_mock` and `create_neptune_offline_mock`) into one. In order to achieve it you'll need to prepare additional side_effect function for `init` that will return
```python
MagicMock(
     __getitem__=MagicMock(side_effect=fetchable_paths), exists=MagicMock(return_value=True)
)
```

based on `mode` argument (`mode=offline` vs non-existence). Then we'll be able to get back to one patch for all of the TestCase class.

Comment (Blaizzy) [reply tests/tests_pytorch/loggers/test_neptune.py:53]: Done ✅

**Review by Raalsky — APPROVED** [review]

**Review by otaj — APPROVED** [review]

**Pull request merged**